### ISSUE TITLE
Where exactly is the "publish to gitHub" option in Team Explorer?

### ISSUE THREAD

**Issue by mortar-and-pestle** [github/VisualStudio#2413]
I hope this is the proper place to ask this as I am new to gitHub and VS extensions. I am using VS 2019 and installed the VS extension. I am logged in and looking at the team explorer. I have my project open, connected with my gitHub, and a repository ready to go on my gitHub. On your website, the instruction states "In the Team Explorer window, click on the Sync tab to get to the Publish to GitHub dialog." However, I see no option on my screen as shown:

![git](https://user-images.githubusercontent.com/36805455/64314999-6f2fb980-cf76-11e9-9f36-da9e6dd8b704.png)

I know it's something obvious but I can't see the publish option under sync. Thanks for any help.

**Comment by Phat6** [comment]
Just figured this out today myself, you need to select changes and add a comment, then commit all. Then select the home icon then sync is on that page, on the bottom you will see a outgoing commits, there is a push there.

**Comment by mortar-and-pestle** [comment]
Thanks. I found the commit. Issue is, it doesn't seem to be actually committing. I have a repository already made. So I opened up my project and followed your instructions. I committed and pushed. What showed up in my gitHub was...not my code. So I deleted it and attempted to do it again and not the commit button is greyed out. So I am not sure what to do here.
![2](https://user-images.githubusercontent.com/36805455/64398002-e32a9a00-d028-11e9-9e3b-4a1c58ee189f.png)

**Comment by Phat6** [comment]
I am really new to Git and trying to figure this out myself. with that said from what I see in your screenshot you are working on Master, you should work on a branch unless this is the first check in. If this is your first check in go to one of your files and add a space or something because it now says you have no changes so you can't check in, you need a change to check in.

**Comment by mortar-and-pestle** [comment]
Yes I am still trying to commit to the master. I added a comment onto one of my files(saved and recompiled) and the commit button is still greyed out. I am thinking about switching to using git commands in the terminal. Thanks for your help though.

**Comment by jcansdale** [comment]
Hi folks 👋 

What you need to do first is `Add to Source Control > Git`.

![image](https://user-images.githubusercontent.com/11719160/64860624-a0b71d80-d625-11e9-8913-c9226df4eba9.png)

You should then see the `Publish to GitHub` options.

![image](https://user-images.githubusercontent.com/11719160/64860679-c47a6380-d625-11e9-9fec-6256a80e775d.png)

In previous versions of Visual Studio, there was an option to automatically add to Source Control when creating a new project. Performing this step afterwards now seems to be the preferred way.

I don't know why Visual Studio shows the Incoming / Outgoing Commits sections when a project isn't in source control. I'd have expected it to show a section inviting the user to `Add to Source Control`.

@daschult do you know if this is a known issue?
